src/Aspire.Dashboard/ConsoleLogs/UrlParser.cs

```diff
@@ -50,17 +52,29 @@ public static bool TryParse(string? text, [NotNullWhen(true)] out string? modifi
 
         modifiedText = null;
         return false;
+
+        static void AppendNonMatchFragment(StringBuilder stringBuilder, Func<string, string>? nonMatchFragmentCallback, string text)
+        {
+            if (nonMatchFragmentCallback != null)
+            {
+                text = nonMatchFragmentCallback(text);
+            }
+
+            stringBuilder.Append(text);
+        }
     }
 
     // Regular expression that detects http/https URLs in a log entry
     // Based on the RegEx used in Windows Terminal for the same purpose, but limited
     // to only http/https URLs
     //
     // Explanation:
-    // /b                             - Match must start at a word boundary (after whitespace or at the start of the text)
+    // (?<=m|\\b)                     - Match must start at a word boundary (after whitespace or at the start of the text) or "m".
+    //                                  "m" is a trailing character of ANSI escape sequences.
+    //                                  Required because URLs are matched before processing ANSI escape sequences.
     // https?://                      - http:// or https://
     // [-A-Za-z0-9+&@#/%?=~_|$!:,.;]* - Any character in the list, matched zero or more times.
     // [A-Za-z0-9+&@#/%=~_|$]         - Any character in the list, matched exactly once
-    [GeneratedRegex("\\bhttps?://[-A-Za-z0-9+&@#/%?=~_|$!:,.;]*[A-Za-z0-9+&@#/%=~_|$]")]
-    private static partial Regex GenerateUrlRegEx();
+    [GeneratedRegex("(?<=m|\\b)https?://[-A-Za-z0-9+&@#/%?=~_|$!:,.;]*[A-Za-z0-9+&@#/%=~_|$]")]
```

Comment:
Looks like we don't use captured groups here, so we could add `RegexOptions.ExplicitCapture` to allocate less when matching, now that we've added a group.

Reply:
I looked at this regex more and I removed the group. I added flags to ignore case and culture.